data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -684,14 +687,80 @@ protected SyncCommitteeRewardData calculateRewards(
+
+              return combinedChainDataClient
+                  .getStateByBlockRoot(block.getRoot())
+                  .thenApply(
+                      maybeState ->
+                          maybeState.map(
+                              state -> getBlockRewardData(blockAndMetaData, block, state)));
+            });
+  }
+
+  private ObjectAndMetaData<BlockRewardData> getBlockRewardData(
+      final BlockAndMetaData blockAndMetaData,
+      final BeaconBlock block,
+      final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state) {
+    if (!spec.atSlot(block.getSlot()).getMilestone().isGreaterThanOrEqualTo(SpecMilestone.ALTAIR)) {
+      throw new BadRequestException(
+          "Slot "
+              + block.getSlot()
+              + " is pre altair, and no sync committee information is available");
+    }
+
+    final UInt64 participantReward = spec.getSyncCommitteeParticipantReward(state);
+    final UInt64 proposerReward =
+        participantReward
+            .times(PROPOSER_WEIGHT)
+            .dividedBy(WEIGHT_DENOMINATOR.minus(PROPOSER_WEIGHT));
+    return blockAndMetaData.map(value -> calculateBlockRewards(proposerReward, block, state));
+  }
+
+  private BlockRewardData calculateBlockRewards(
+      final UInt64 proposerReward,
+      final BeaconBlock block,
+      final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state) {
+    final SyncAggregate aggregate = block.getBody().getOptionalSyncAggregate().orElseThrow();
+
+    final int proposerIndex = spec.getBeaconProposerIndex(state, block.getSlot());
+    final long attestationsBlockRewards = calculateAttestationRewards(state);
+    final UInt64 syncAggregateBlockRewards =
+        calculateProposerSyncAggregateBlockRewards(proposerReward, aggregate);
+    final UInt64 proposerSlashingsBlockRewards = calculateProposerSlashingsRewards(block, state);
+    final UInt64 attesterSlashingsBlockRewards = calculateAttesterSlashingsRewards(block, state);
+    final long total =
+        attestationsBlockRewards
+            + syncAggregateBlockRewards.longValue()
+            + proposerSlashingsBlockRewards.longValue()
+            + attesterSlashingsBlockRewards.longValue();
+
+    return new BlockRewardData(
+        proposerIndex,
+        total,
+        attestationsBlockRewards,
+        syncAggregateBlockRewards,
+        proposerSlashingsBlockRewards,
+        attesterSlashingsBlockRewards);
+  }
+
   @VisibleForTesting
-  protected int calculateProposerSyncAggregateBlockRewards(
+  protected UInt64 calculateProposerSyncAggregateBlockRewards(
       UInt64 proposerReward, SyncAggregate aggregate) {
     final SszBitvector syncCommitteeBits = aggregate.getSyncCommitteeBits();
-    int total = 0;
+    UInt64 total = ZERO;
     for (int i = 0; i < syncCommitteeBits.size(); i++) {
       if (syncCommitteeBits.getBit(i)) {
-        total += proposerReward.intValue();
+        total = total.plus(proposerReward);
       }
```

Comment:
so really here we're counting the bits set, which we could just do with an int count, then `return proposerReward.times(count)`